### PR TITLE
RC1 v1.2.4

### PR DESCRIPTION
**Pull request by adam-hrin-solteq**
New release v1.2.4

### CODE REVIEW

File: Helper/Version.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public function getDecodedContentFromGithub()
+    {
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, self::GIT_URL);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 1);
+        curl_setopt($ch, CURLOPT_USERAGENT, 'magento');
+        $content = curl_exec($ch);
+        curl_close($ch);
+        return json_decode($content, true);
```
Comment (MAleksei) [inline Helper/Version.php:48]: Magento has own curl methods to use. should we change to that ?
https://devdocs.magento.com/guides/v2.3/get-started/gs-curl.html

File: Model/ReceiptDataProvider.php [hunk 2]
```diff
@@ -206,7 +212,8 @@ public function __construct(
         OrderInterface $orderInterface,
         transactionBuilder $transactionBuilder,
         Config $gatewayConfig,
-        ApiData $apiData
+        ApiData $apiData,
+        LoggerInterface $logger
```
Comment (MAleksei) [inline Model/ReceiptDataProvider.php:216]: di compile going through with fine ?
context might have already instance of logger interface.
$context->getLogger(); // return \Psr\Log\LoggerInterface

File: Helper/Version.php (hunk 1 above)
Comment (adam-hrin-solteq) [reply Helper/Version.php]: the code is now changed to use the magento curl

File: Model/ReceiptDataProvider.php (hunk 2 above)
Comment (adam-hrin-solteq) [reply Model/ReceiptDataProvider.php:216]: the context does not seem to have the logger, di compile is going through ok

**Review by MAleksei — APPROVED** [review]
this look nice.

**Pull request merged**